SAML2/XML/Signature.hs

```diff
@@ -247,8 +249,21 @@ _verifySignatureOld pks xid doc = do
   xpathbase = "/*" ++ xpathsel "Signature" ++ xpathsel "SignedInfo" ++ "//"
   xpath = xpathbase ++ ". | " ++ xpathbase ++ "@* | " ++ xpathbase ++ "namespace::*"
 
+-- | It turns out sometimes we don't get envelopped signatures, but signatures that are
+-- located outside the signed sub-tree.  Since 'verifySiganture' doesn't support this case, if
+-- you encounter it you should fall back to 'verifySignatureLegacy'.
+verifySignatureLegacy :: PublicKeys -> String -> HXT.XmlTree -> IO (Either SignatureError ())
```

Comment:
Should `verifySignature` be removed from exports or removed altogether?

Reply:
Possibly?  I'll rename it to "incomplete" for now, that will keep it available for hacking.